src/renderer/renderElement.ts

```diff
@@ -93,6 +93,42 @@ export interface ExcalidrawElementWithCanvas {
   boundTextElementVersion: number | null;
 }
 
+const cappedElementCanvasSize = (
+  element: NonDeletedExcalidrawElement,
+  zoom: Zoom,
+): {
+  width: number;
+  height: number;
+  zoomValue: NormalizedZoomValue;
+} => {
+  const sizelimit = 16777216; // 2^24
```

Comment:
where's the 16M limit for safari coming from? MDN and SO says different values. 16M maybe iOS? But even there.

If we got this low, I'd increase it for other browsers. Though we don't have to go to their limits. Will at least improve memory and perf in general.

That said, we should also cap width & height individually as there are limits on those irrespective of the area.

Reply:
the 16 came from a warning message when I was debugging Excalidraw on iOS. We can increase it for other browsers, though as you say, I also found that with the 16M limit, the images looked ok on all my devices.

Reply:
I updated the comment in the code
Here's a [blogpost](https://pqina.nl/blog/canvas-area-exceeds-the-maximum-limit/#:~:text=It%27s%202022%20and%20Safari%20still,216%20pixels.)  that includes the warning I also saw while debugging excalidraw.

![](https://pqina.nl/media/safari-canvas-area-warning.png)